ntbs-service/DataAccess/NotificationRepository.cs

```diff
@@ -146,5 +147,12 @@ public async Task<Notification> GetNotificationWithAllInfoAsync(int? Notificatio
         public IQueryable<Notification> GetBaseQueryableNotificationByStatus(IList<NotificationStatus> statuses) {
             return GetBaseNotificationIQueryable().Where(n => statuses.Contains(n.NotificationStatus));
         }
+
+        public Task<List<Notification>> GetNotificationsWithDetailsById(IList<int> ids)
```

Comment:
I think method declaration would be better as:
`public Task<IList<Notification>> GetNotificationsByIdsAsync(IList<int> ids)`
(IList + Async + Ids)
Don't know what I would prefer here regarding 'Details' or not - if included I think it should specific which details are included.